### DISCUSSION TITLE
odmfuse.sh in generated tarball

### DISCUSSION THREAD

**Discussion by mbilloo** [OE4T/meta-tegra#698] [Q&A] [answered]
Hi,

From the wiki  page [Secure Boot Support](https://github.com/OE4T/meta-tegra/wiki/Secure-Boot-Support), it says to use odmfuse.sh in L4T to burn the fuses. Does that mean it is not included as part of the generated tarball in the Yocto build (I'm not seeing it and just wanted to make sure that my observation is correct)?

**Comment by madisongh** [comment, marked as answer]
That's right.

**Reply by mbilloo** [reply]
Just out of curiosity and my own edifcation, why was it not included?

**Reply by madisongh** [reply]
A few reasons:
* It's not really related to building/deploying the software, since fusing is a separate process
* The NVIDIA scripts tend to assume that you're running them in their L4T tree. Testing that this particular script works after we've repackaged it is not exactly easy, given what it does
* The script already has the `--no-burn` option for generating a self-contained tarball for programming the fuses without having the full L4T kit installed

There is some support in the layer where you can supply the `odmfuse_pkc.xml` file generated by the `odmfuse.sh --no-burn` option and have it be bundled in the `tegraflash` package, and a script will be generated to perform the fuse burning using that XML file.  It needs a bit more work, though, to make it possible to use a single `tegraflash` package to do both fuse burning and flashing with signed binaries.  And it needs some documentation, too. :)
